### ISSUE TITLE
VSCode extension crashes the extension host at extension activation

### ISSUE THREAD

**Issue by schl3ck** [quick-lint/quick-lint-js#1182]
When the extension is activated the extension host immediately crashes. It doesn't matter what .js file I've open.

The only clue I found was in the developer tools of VSCode. All other output views did not provide any useful information. In the console of the developer tools it logged:

```
D:/a/quick-lint-js/quick-lint-js/src/quick-lint-js/io/file-canonical.cpp:676: internal check failed in resolve_symlink: false
quick-lint-js crashed. Please report this bug here:
https://quick-lint-js.com/crash-report/
```

Please let me know if there is any other information I can provide.

**Comment by strager** [comment]
Thanks for the bug report. I'll try to reproduce the issue this weekend.

**Comment by strager** [comment]
I can reproduce the issue if I create a directory junction and open a file through that junction. For example:
```
mklink /j p:\quick-lint-js-sandbox p:\quick-lint-js\sandbox
code p:\quick-lint-js-sandbox\hello.js
```
I could not reproduce this issue when using subst or a network drive.

@schl3ck Are you using a junction/symlink?

**Comment by schl3ck** [comment]
Yes, I'm editing the files through a junction.

**Comment by strager** [comment]
Fixed in Git commit 748b775a60ae558c8d717d2be1e6e58f80b0cae5. This fix will be included in quick-lint-js version 3.2.

**Comment by strager** [comment]
Fix released in [version 3.2.0](https://quick-lint-js.com/releases/#3.2.0).
